embeddings/pipeline/flair_preprocessing_pipeline.py

```diff
@@ -89,17 +93,30 @@ def _get_dataloader(self, dataset: Dataset) -> FLAIR_DATALOADERS:
 
     def _get_dataset_transformation(
         self, data_loader: FLAIR_DATALOADERS
-    ) -> FLAIR_DATASET_TRANSFORMATIONS_TYPE:
+    ) -> Optional[FLAIR_DATASET_TRANSFORMATIONS_TYPE]:
         if isinstance(data_loader, (ConllFlairCorpusDataLoader, PickleFlairCorpusDataLoader)):
-            return DummyTransformation()
+            return None
 
-        return self._get_base_dataset_transformation()
+        return self._get_to_flair_dataset_transformation()
 
     def _get_transformations(
         self, data_loader: FLAIR_DATALOADERS
     ) -> FLAIR_DATASET_TRANSFORMATIONS_TYPE:
 
-        transformation = self._get_dataset_transformation(data_loader)
+        transformation: FLAIR_DATASET_TRANSFORMATIONS_TYPE = DummyTransformation()
+        if self.encode_labels:
```

Comment:
consider early exit, without extra nested if?
if self.encode_labels and isinstance(data_loader, (ConllFlairCorpusDataLoader, PickleFlairCorpusDataLoader)):
                raise ValueError(
                    "ClassEncodeColumnTransformation transformation is unavailable for Flair DataLoaders. "
                    "Set parameter `encode_labels` value to True"
                )
transformation= transformation.then(
                ClassEncodeColumnTransformation(column=self.target_column_name)
            ) if self.encode_labels else transformation